core/src/main/scala/pureconfig/error/FailureReason.scala

```diff
@@ -158,3 +158,18 @@ final case class WrongSizeString(expected: Int, found: Int) extends FailureReaso
 final case class NoValidCoproductChoiceFound(value: ConfigValue) extends FailureReason {
   def description = s"No valid coproduct choice found for '${value.render(ConfigRenderOptions.concise())}'."
 }
+
+/**
+ * A failure reason given when a ConfigObject can't be converted to a ConfigList.
+ *
+ * @param failingKeys the list of keys whose conversion to integer failed
+ */
+final case class CannotConvertObjectToList(failingKeys: List[String]) extends FailureReason {
```

Comment:
Do you think it's worth it to add a specific `CannotConvert`-like class for the conversion between objects and lists? We've been using `CannotConvert` for all type conversion errors throughout the codebase... I don't know if there are other cases with conversion-specific data to provide like this one.

HOCON clearly fuzzies the separation of various types in their AST with their auto-conversions. As a stretchy interpretation, I'd be tempted to consider the type of an object with numeric keys as both a list and an object and use `WrongType` here by saying that we were expecting a list but got an object.

Reply:
I personally never use `FailureReason`s directly, but I think it's a good idea to provide as much actionable information as possible when we fail reading something. I haven't gone through all usages of `CannotConvert` in the codebase, but most of them build upon exception messages, from which we can't extract much.

That being said, I'm also OK with using `CannotConvert` with a nice message.